### PR TITLE
feat: display catalogs and baselines

### LINKED ISSUE #887: Display Existing Catalogs and Baselines (Profiles)
### Description

As a security practitioner, I want to see a list of all OSCAL-based catalogs and baselines (profiles) within the system and be able to select one for viewing/editing.

The list should include the following for each OSCAL catalog or profile within the system:
- Title
- Last Modified
- Document Version
- Publication Date

Each item should also offer the ability to open the catalog or profile. 
The detail of what happens when a catalog or profile is open is outside the scope of this issue. 

### Acceptance Criteria

- Aligned with the Spring 2023 styling strategy, guidance, and standards
- The user is able to indicate they wish to see a list of catalogs and baselines
- A list of all available catalogs and baselines is presented to the user
- The list includes:
  - Title
  - Last Modified
  - Document Version
  - Publication Date

### Proposed Solution

- A transaction may need to be defined on the back-end to support this call. 
- Perform an a Transaction API call to query for the list
- Format and present the list to users

### Additional Notes

See the screen mockup and UI developer's notes in [Figma](https://www.figma.com/file/ntIfIHIhMYbiqS9Nuc7EQG/Catalog-Mockups?type=design&node-id=8-1329&t=x5GqLMfITPmAInmo-0)

### Implementation

- [ ] I can (or plan to) submit a pull request to implement this
- [ ] Implementing this may result in a breaking change

### PR DESCRIPTION
**Pull request by mpemy**
Closes #887

### CODE REVIEW

File: packages/oscal-react-library/src/components/OSCALCatalogGroup.tsx [hunk 1]
```diff
@@ -174,6 +175,17 @@ export interface OSCALCatalogControlListItemProps
...
+/**
+ * Creates a new catalog group using catalog metadata info
+ 
+export const OSCALCreateCatalogGroup: React.FC<OSCALModelMetadataInfo> = (item) => {
+  return (
+    <>
+      <CatalogBreadCrumbsMenu title={item.title}></CatalogBreadCrumbsMenu>
+    </>
+  );
+};
```
Comment (Bronstrom) [inline packages/oscal-react-library/src/components/OSCALCatalogGroup.tsx:187]: I am noticing a bit of commented code out in places, such as this function here. I was just curious if these were mainly being used for trial-and-error testing or these are these are places of functionality you plan to add and they aren't working quite yet.

**Review by Bronstrom — COMMENTED** [review]
Good work so far on this @mpemy! I have a couple styling/refactoring comments and left a question or two below:

File: packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx [hunk 2]
```diff
@@ -0,0 +1,1512 @@
...
+      contact.email = event.target.value;
+    }
+    function handleEditCityChange(event: { target: { value: string | undefined } }) {
+      address.city = event.target.value;
+    }
+
+    function handleEditStateChange(event: { target: { value: string | undefined } }) {
+      address.state = event.target.value;
+    }
+    function handleEditZipChange(event: { target: { value: string | undefined } }) {
```
[the remark this answers is not in the document]
Comment (mpemy) [reply packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx:889]: Thanks @Bronstrom you are right, I will refactor this, these two dialogs are quite similar. Thanks for your insight.

File: packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx [hunk 3]
```diff
@@ -0,0 +1,1512 @@
...
+  right: 62.27%;
+  top: 32.69%;
+  bottom: 70.31%;
+  font-family: "Source Sans Pro";
+  font-style: normal;
+  font-weight: 400;
+  font-size: 14px;
+  line-height: 18px;
+  color: #2b2b2b;
+`;
```
[the remark this answers is not in the document]
Comment (mpemy) [reply packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx:117]: @Bronstrom You are right, all these CSS directives should be handled in the Global Style. But in order to complete the feature I had to add those directives, but once Global Style is implemented, I will have to readjust this implementation to directly inherit the Global Style.
I removed all mention of _font-style: normal_ and _font-size: 16px_

File: packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx [hunk 4]
```diff
@@ -0,0 +1,1512 @@
...
+  font-size: 20px;
+  line-height: 25px;
+  color: #002867;
+`;
+const LastModified = styled(Typography)`
+  position: absolute;
+  height: 18px;
+  width: 350px;
+  left: 332.375px;
+  top: 249.21400451660156px;
```
[the remark this answers is not in the document]
Comment (mpemy) [reply packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx:105]: @Bronstrom Thanks for point this out, I copied and pasted from the Figma Mockups, that is why it had those extremely decimal value. I rounded those values accordingly.

File: packages/oscal-react-library/src/components/OSCALCatalogGroup.tsx (hunk 1 above)
Comment (mpemy) [reply packages/oscal-react-library/src/components/OSCALCatalogGroup.tsx:186]: @Bronstrom, I addressed all review recommendations and suggestions. Thanks!

File: packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx [hunk 5]
```diff
@@ -0,0 +1,1518 @@
...
+                },
+              }}
+            />
+            <CTab
+              label="Catalog Details"
+              value="2"
+              sx={{
+                ":focus": {
+                  background: "#FFffff",
+                  weight: 600,
```
Comment (Bronstrom) [inline packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx:432]: This sx `focus` styling also looks common to `CTabs`. Can this also be placed in the styled component above?

File: packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx [hunk 6]
```diff
@@ -0,0 +1,1518 @@
...
+              onClick={handleAddNewCatalogBaseline}
+            >
+              PUBLISH
+            </OSCALSecondaryButton>
+          </Grid>
+        </Hug>
+      );
+    }
+  }
+  class HugHug extends React.Component {
```
Comment (Bronstrom) [inline packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx:586]: Two things here:
1) We haven't been using `class` throughout the application anywhere else except for some of the type objects Kyle had created. To make things consistent could we turn all `class`es into `function`s or `const`s for component rendering code like this? We can remove the `extends React.Component` part and `render()` function this way.
2) A bit of a nitpick, but could we name this something a little more description than "HugHug", maybe something focused around the `Create New` button handling a new catalog or baseline.

File: packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx [hunk 7]
```diff
@@ -0,0 +1,1518 @@
...
+  top: 40px;
+  left: 10px;
+  font-family: Source Sans Pro;
+  font-size: 20px;
+  font-weight: 700;
+  line-height: 25px;
+  letter-spacing: 0em;
+  text-align: left;
+  background: #ffffff;
+`;
```
[the remark this answers is not in the document]
Comment (Bronstrom) [reply packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx:209]: I don't mind handling this in the styling issue for displaying catalogs and baselines.

**Review by Bronstrom — APPROVED** [review]
This looks great to me @mpemy! Thanks for your hard work on this issue.

**Pull request merged**